caravel/assets/javascripts/SqlLab/components/DataPreviewModal.jsx

```diff
@@ -0,0 +1,58 @@
+import * as Actions from '../actions';
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Modal } from 'react-bootstrap';
+
+import ResultSet from './ResultSet';
+
+const propTypes = {
+  queries: React.PropTypes.array,
+  actions: React.PropTypes.object,
+  showDataPreviewModal: React.PropTypes.bool,
+  dataPreviewQueryId: React.PropTypes.string,
```

Comment:
this comes straight from the redux state, I guess I could have some other wrapper component that would control all this, but that adds a level in the hierachy